homeassistant/components/plugwise/__init__.py

```diff
@@ -1 +1,170 @@
+        _LOGGER.error("Error while communicating to device")
+        raise ConfigEntryNotReady
+
+    except asyncio.TimeoutError:
+        _LOGGER.error("Timeout while connecting to Smile")
+        raise ConfigEntryNotReady
+
+    if api.smile_type == "power":
+        update_interval = timedelta(seconds=10)
+    else:
+        update_interval = timedelta(seconds=60)
+
+    async def async_update_data():
+        """Update data via API endpoint."""
+        try:
+            async with async_timeout.timeout(10):
+                await api.full_update_device()
+                return True
+        except Smile.XMLDataMissingError:
+            raise UpdateFailed("Smile update failed")
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="Smile",
+        update_method=async_update_data,
+        update_interval=update_interval,
+    )
+
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    api.get_all_devices()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
+        "api": api,
+        "coordinator": coordinator,
+    }
+
+    device_registry = await dr.async_get_registry(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        identifiers={(DOMAIN, api.gateway_id)},
+        manufacturer="Plugwise",
+        name=entry.title,
+        model=f"Smile {api.smile_name}",
+        sw_version=api.smile_version[0],
+    )
+
+    # single_master_thermostat = api.single_master_thermostat()
+    platforms = ALL_PLATFORMS
+    # if single_master_thermostat is None:
+    #     platforms = SENSOR_PLATFORMS
+
+    for component in platforms:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in ALL_PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class SmileGateway(Entity):
+    """Represent Smile Gateway."""
+
+    def __init__(self, api, coordinator):
+        """Initialise the sensor."""
+        self._api = api
+        self._coordinator = coordinator
+
+    @property
+    def should_poll(self):
+        """Return False, updates are controlled via coordinator."""
+        return False
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._coordinator.last_update_success
+
+    async def async_added_to_hass(self):
+        """Subscribe to updates."""
+        self.async_on_remove(
+            self._coordinator.async_add_listener(self._update_callback)
+        )
```

Comment:
```suggestion
```

Ideally all you should need to do is `self.async_write_ha_state` and all the properties will consume the data from the coordinator when Home Assistant processes the state change.

Reply:
I tried without it, reviewing some other platforms, but our `def update(self):` would never be called. With adding this back in, as per [this component](https://github.com/home-assistant/core/blob/7ac547a6e03693a3e3b19da3b131035fc6c866ce/homeassistant/components/nws/weather.py#L115) it worked.

Reply:
You can resolve that in two possible way.

1. Move the code in `def update()` that recalculates each property from the coordinator data into the properties themselves
2. Move the code in `def update()` to `def _update_callback()`, delete the `self._coordinator.async_add_listener(self.async_write_ha_state)`, and delete `def update()` as we shouldn't have both `async_update()` and `update()` in the same class.

Reply:
I opted for the second option

Reply:
I thought you might :)